### PR TITLE
Map Predicates

### PR DESCRIPTION
**Pull request by mustafaiman**
I added new methods with `withPredicate` suffix.

Making predicate optional in `addEntryListener` complicates the API very much. Because I added a new method for addEntryListener with predicate, I added new methods for others too for the sake of uniformity.

### CODE REVIEW

**Comment by GhostInAMachine** [comment]
Can you split the map tests into smaller files? It's eight hundred lines already

File: src/proxy/MapProxy.ts [hunk 1]
```diff
@@ -45,7 +45,59 @@ import {EntryEventType} from '../core/EntryEventType';
...
+import {MapAddEntryListenerToKeyWithPredicateCodec} from '../codec/MapAddEntryListenerToKeyWithPredicateCodec';
+import {MapAddEntryListenerWithPredicateCodec} from '../codec/MapAddEntryListenerWithPredicateCodec';
 export class MapProxy<K, V> extends BaseProxy implements IMap<K, V> {
+
+    entrySetWithPredicate(predicate: Predicate<K, V>): Promise<any[]> {
+        assertNotNull(predicate);
+        var predicateData = this.toData(predicate);
+        var toObject = this.toObject.bind(this);
+        var deserializedSet: [K, V][] = [];
+        return this.encodeInvokeOnRandomTarget(MapEntriesWithPredicateCodec, predicateData)
```
Comment (GhostInAMachine) [inline src/proxy/MapProxy.ts]: Fix alignment here

File: src/proxy/MapProxy.ts [hunk 2]
```diff
@@ -45,7 +45,59 @@ import {EntryEventType} from '../core/EntryEventType';
...
+            return deserializedSet;
+        });
+    }
+
+    keySetWithPredicate(predicate: Predicate<K, V>): Promise<K[]> {
+        assertNotNull(predicate);
+        var predicateData = this.toData(predicate);
+        var toObject = this.toObject.bind(this);
+        var deserializedSet: K[] = [];
+        return this.encodeInvokeOnRandomTarget(MapKeySetWithPredicateCodec, predicateData).then(function (entrySet: Data[]) {
```
Comment (GhostInAMachine) [inline src/proxy/MapProxy.ts]: Perhaps you can replace this with `map` calls?

**Comment by mustafaiman** [comment]
We can split Map tests with another PR. I do not want to mix it with this one.

**Pull request merged**